api/src/routes/category-list.ts

```diff
@@ -19,7 +24,7 @@ export async function categoryListHandler(req: Request, res: Response) {
         description: true,
         createdAt: true,
       },
-      where: (category, { eq }) => eq(category.shopId, user.shopId),
+      where: (category, { eq }) => eq(category.shopId, shopId),
```

Comment:
We only need the shop id for querying, so we can use `getShopIdBySite` instead to simplify things.